test/fixtures/specs/stale.spec.js

```diff
@@ -23,4 +23,24 @@ describe('staleElementRetry', () => {
         browser.url(conf.testPage.staleTest)
         browser.staleMe(100)
     })
+
+    it('does not throw staleElementReference exception when waiting for element to become invisible but which is removed from DOM in a custom command', () => {
+
+        browser.addCommand('waitForInvisibleInParallel', (iterations = 20) => {
+            const promises = []
+            for (let i = 1; i <= iterations; i++) {
+                promises.push(browser.waitForVisible('.staleElementContainer2 .stale-element-container-row-' + i, 10000, true))
```

Comment:
`browser.waitForVisible()` does not return a promise here. You have to add async as function name to run in a promise context